src/installer/installer.go

```diff
@@ -346,13 +346,14 @@ func (i *installer) downloadHostIgnition() (string, error) {
 }
 
 func (i *installer) waitForNetworkType(kc k8s_client.K8SClient) error {
-	return utils.WaitForPredicate(waitForeverTimeout, 5*time.Second, func() bool {
+	err, _, _ := Wait().WithNoTimeout().WithInterval(5 * time.Second).WithPredicate(func() bool {
```

Comment:
why without timeout?

Reply:
i am not sure i really like the usage i must say. In any case i think it should start with WithPredicate and then adding with interval and etc. 
maybe we can just move to https://github.com/avast/retry-go ?